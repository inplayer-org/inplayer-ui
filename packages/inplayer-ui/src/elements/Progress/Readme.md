```jsx static
import { Progress } from '@inplayer-org/inplayer-ui';

function Page(props) {
  return (
    <Progress
      type="circle"
      percent="10"
      strokeWidth="4"
      strokeColor="#D3D3D3"
    />
  );
}
```

### Demo

```jsx
const containerStyle = {
  width: '250px',
};
const circleContainerStyle = {
  width: '250px',
  height: '250px',
  display: 'inline-block',
};

<div>
  <h3>Line Progress 10%</h3>
  <div style={containerStyle}>
    <Progress type="line" percent="10" strokeWidth="2" />
  </div>
  <h3>Circle Progress 40%</h3>
  <div style={circleContainerStyle}>
    <Progress type="circle" percent="40" strokeWidth="3" />
  </div>
</div>;
```
