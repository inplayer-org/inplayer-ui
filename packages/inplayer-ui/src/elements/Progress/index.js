// @flow
import React from 'react';
import { Line, Circle } from 'rc-progress';

type ProgressType = 'circle' | 'line';

type Props = {
  type?: ProgressType,
  /** Width of the stroke. Unit is percentage of SVG canvas size.  */
  strokeWidth?: number,
  /** Stroke color. */
  strokeColor?: string,
  /** Width of the trail stroke. Unit is percentage of SVG canvas size. Trail is always centered relative to actual progress path. If trailWidth are not defined, it same as strokeWidth.  */
  trailWidth?: number,
  /** Color for lighter trail stroke underneath the actual progress path. */
  trailColor?: string,
  /** The shape to be used at the end of the progress bar, can be `butt`, `square` or `round`. */
  strokeLinecap?: string,
  /** prefix className for component */
  prefixCls?: string,
  /** customized className */
  className?: string,
  /** style object will be added to svg element  */
  style?: Object,
  /** the percent of the progress  */
  percent?: number,
  /** the gap degree of half circle, 0 - 360  */
  gapDegree?: number,
  /** the gap position, value: top, bottom, left, right.  */
  gapPosition?: string,
};

const Progress = ({ type, ...rest }: Props) =>
  type === 'circle' ? <Circle {...rest} /> : <Line {...rest} />;

Progress.defaultProps = {
  type: 'circle',
  strokeWidth: 1,
  strokeColor: '#07AAE6',
  trailWidth: 1,
  trailColor: '#d7dde5',
  strokeLinecap: 'round',
  prefixCls: 'rc-progress',
  className: '',
  percent: 0,
  gapDegree: 0,
  gapPosition: 'top',
  style: {},
};

/** @component */
export default Progress;
