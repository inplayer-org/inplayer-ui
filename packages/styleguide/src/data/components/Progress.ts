import { ComponentData } from "../types";

const usage = `
  import { Progress } from '@inplayer-org/inplayer-ui';

  function Page(props) {
    return (
      <Progress
        type="circle"
        percent="10"
        strokeWidth="4"
        strokeColor="#D3D3D3"
      />
    );
  }
`;

const demoCode = `
  <div>
    <h3>Line Progress 10%</h3>
    <div style={{width: '250px'}}>
      <Progress type="line" percent="10" strokeWidth="2" />
    </div>
    <h3>Circle Progress 40%</h3>
    <div style={{width: '250px',height: '250px', display: 'inline-block'}}>
      <Progress type="circle" percent="40" strokeWidth="3" />
    </div>
  </div>;
`

const Progress: ComponentData = {
  description: ``,
  path: `src/elements/Progress/index.js`,
  usage,
  examples: [
    {
      title: 'Demo',
      code: demoCode,
    }
  ],
  propsAndMethods: [
    { name: 'className', type: 'string', description: 'customized className' },
    { name: 'gapDegree', type: 'number', default: '0', description: 'the gap degree of half circle, 0 - 360' },
    { name: 'gapPosition', type: 'string', default: 'top' , description: 'the gap position, value: top, bottom, left, right.' },
    { name: 'percent', type: 'number', default: '0', description: 'the percent of the progress' },
    { name: 'prefixCls', type: 'string', default: 'rc-progress', description: 'A className can be passed down for further styling or extending with CSS-in-JS'},
    { name: 'strokeColor', type: 'string', default: '#07AAE6', description: 'Stroke color' },
    { name: 'strokeLinecap', type:'string', default: 'round', description: 'The shape to be used at the end of the progress bar, can be butt, square or round.' },
    { name: 'strokeWidth', type:'number', default: '1', description: 'Width of the stroke. Unit is percentage of SVG canvas size.' },
    { name: 'style', type:'Object', default: '{}', description: 'style object will be added to svg element' },
    { name: 'trailColor', type:'string', default: '#d7dde5', description: 'Color for lighter trail stroke underneath the actual progress path' },
    { name: 'trailWidth', type: 'number', default:'1', description: 'Width of the trail stroke. Unit is percentage of SVG canvas size. Trail is always centered relative to actual progress path. If trailWidth are not defined, it same as strokeWidth.' },
    { name: 'type', type: 'enum', default:'circle', description: 'One of: circle, line' },
  ]
}

export default Progress;