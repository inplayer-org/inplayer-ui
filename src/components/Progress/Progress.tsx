import React from 'react';
import { Line, Circle } from 'rc-progress';

type ProgressType = 'circle' | 'line';

type Type = {
  /**
   * Progress type
   */
  type?: ProgressType;
};

type RCProgressProps = {
  /**
   * Width of the stroke. Unit is percentage of SVG canvas size
   */
  strokeWidth?: number;
  /**
   * Width of the trail stroke. Unit is percentage of SVG canvas size. Trail is always centered relative to actual progress path. If trailWidth are not defined, it same as strokeWidth.
   */
  trailWidth?: number;
  /**
   * The percent of the progress
   */
  percent?: number | number[];
  /**
   * Stroke color
   */
  strokeColor?: string | string[] | object;
  /**
   * Color for lighter trail stroke underneath the actual progress path
   */
  trailColor?: string;
  /**
   * The shape to be used at the end of the progress bar, can be `butt`, `square` or `round`.
   */
  strokeLinecap?: 'butt' | 'square' | 'round';
  /**
   * Prefix className for component
   */
  prefixCls?: string;
  /**
   * The gap degree of half circle, 0 - 360
   */
  gapDegree?: number;
  /**
   * The gap position, value: top, bottom, left, right.
   */
  gapPosition?: 'top' | 'right' | 'bottom' | 'left';
};

type Props = Type & RCProgressProps;

const Progress = ({ type, ...rest }: Props) =>
  type === 'circle' ? <Circle {...rest} /> : <Line {...rest} />;

Progress.defaultProps = {
  type: 'circle',
  strokeWidth: 1,
  strokeColor: '#07AAE6',
  trailWidth: 1,
  trailColor: '#d7dde5',
  strokeLinecap: 'round',
  prefixCls: 'rc-progress',
  percent: 0,
  gapDegree: 0,
  gapPosition: 'top',
};

export default Progress;
