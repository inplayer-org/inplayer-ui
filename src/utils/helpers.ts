import moment from 'moment';

export const getMonthOptions = (): Array<any> =>
  moment.months().map((label, value) => ({ value, displayName: label }));

export const getYearOptions = () => {
  const yearOptions: Array<any> = [];
  for (let i = moment().year() - 100; i <= moment().year(); i += 1) {
    yearOptions.push({ value: i.toString(), displayName: i.toString() });
  }

  return yearOptions;
};
