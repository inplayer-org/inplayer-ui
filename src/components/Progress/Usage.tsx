import React from 'react';
import Progress from '.';

const containerStyle = {
  width: '250px',
};
const circleContainerStyle = {
  width: '250px',
  height: '250px',
  display: 'inline-block',
};

const Usage = () => (
  <div>
    <h3>Line Progress 10%</h3>
    <div style={containerStyle}>
      <Progress type="line" percent={10} strokeWidth={3} strokeColor="red" trailColor="black" />
    </div>
    <h3>Circle Progress 40%</h3>
    <div style={circleContainerStyle}>
      <Progress type="circle" percent={40} strokeWidth={3} strokeColor="green" />
    </div>
  </div>
);

export default Usage;
