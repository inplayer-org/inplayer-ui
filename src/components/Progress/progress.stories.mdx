import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import Progress from './'

<Meta title="Progress" component={<Progress/>} />

# Progress

```jsx
import { Progress } from '@inplayer-org/inplayer-ui';

const Page = () => (
  <Progress />
)
```
### Props

<Props of={Progress}/>

### Line Progress
<Preview withSource="open">
  <Story name="line">
  {() => {
    const containerStyle = {
        width: '250px',
    };
    return(
    <div>
        <h3>Line Progress 10%</h3>
        <div style={containerStyle}>
        <Progress type="line" percent="10" strokeWidth="2" />
        </div>
    </div>
    )
  }}
  </Story>
</Preview>

### Circle Progress
<Preview withSource="open">
  <Story name="circle">
  {() => {
    const containerStyle = {
        width: '250px',
    };
    return(
    <div>
        <h3>Circle Progress 40%</h3>
        <div style={containerStyle}>
        <Progress type="circle" percent="40" strokeWidth="3" />
        </div>
    </div>
    )
  }}
  </Story>
</Preview>
