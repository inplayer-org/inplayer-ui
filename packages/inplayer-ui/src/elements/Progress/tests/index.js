import { generateSnapshots } from 'packages/inplayer-ui/src/elements/Progress/tests/helpers';

import Progress from '../index';

generateSnapshots(Progress);
