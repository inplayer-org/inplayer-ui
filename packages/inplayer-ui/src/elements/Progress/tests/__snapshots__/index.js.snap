// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders correctly 1`] = `
<svg
  className="rc-progress-circle "
  style={Object {}}
  viewBox="0 0 100 100"
>
  <path
    className="rc-progress-circle-trail"
    d="M 50,50 m 0,-49.5
   a 49.5,49.5 0 1 1 0,99
   a 49.5,49.5 0 1 1 0,-99"
    fillOpacity="0"
    stroke="#d7dde5"
    strokeLinecap="round"
    strokeWidth={1}
    style={
      Object {
        "stroke": "#d7dde5",
        "strokeDasharray": "311.01767270538954px 311.01767270538954px",
        "strokeDashoffset": "-0px",
        "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
      }
    }
  />
  <path
    className="rc-progress-circle-path"
    d="M 50,50 m 0,-49.5
   a 49.5,49.5 0 1 1 0,99
   a 49.5,49.5 0 1 1 0,-99"
    fillOpacity="0"
    stroke=""
    strokeLinecap="round"
    strokeWidth={0}
    style={
      Object {
        "stroke": "#07AAE6",
        "strokeDasharray": "0px 311.01767270538954px",
        "strokeDashoffset": "-0px",
        "transition": "stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s",
      }
    }
  />
</svg>
`;
